packages/@aws-cdk/aws-kinesisfirehose-destinations/lib/common.ts

```diff
@@ -1,6 +1,43 @@
 import * as iam from '@aws-cdk/aws-iam';
 import * as logs from '@aws-cdk/aws-logs';
 
+/**
+ * Possible compression options Kinesis Data Firehose can use to compress data on delivery.
+ */
+export class Compression {
+  /**
+   * gzip
+   */
+  public static readonly GZIP = new Compression('GZIP');
+
+  /**
+   * Hadoop-compatible Snappy
+   */
+  public static readonly HADOOP_SNAPPY = new Compression('HADOOP_SNAPPY');
+
+  /**
+   * Snappy
+   */
+  public static readonly SNAPPY = new Compression('Snappy');
+
+  /**
+   * Uncompressed
+   */
+  public static readonly UNCOMPRESSED = new Compression('UNCOMPRESSED');
```

Comment:
How about we just leave this one out, and `undefined` means uncompressed?

We generally don't represent "empty" with a special value if the default is "empty".